src/Host/Broker/Impl/Security/TlsConfiguration.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Net.Security;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Server.Kestrel.Https;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.R.Host.Protocol;
+
+namespace Microsoft.R.Host.Broker.Security {
+    internal sealed class TlsConfiguration {
+        private readonly ILogger _logger;
+
+        public TlsConfiguration(ILogger logger) {
```

Comment:
This should probably take `SecurityOptions` in constructor, since it's going to be the same for all calls on it.